flood_adapt/adapter/direct_impacts_integrator.py

```diff
@@ -155,97 +175,99 @@ def preprocess_fiat(self):
+                )
+                dem = (
+                    self.database.static_path
+                    / "dem"
+                    / self.site_info.attrs.dem.filename
                 )
-            elif measure.attrs.type == "floodproof_properties":
-                fa.floodproof_properties(
-                    floodproof=measure,
+                aggregation_areas = [
+                    self.database.static_path / aggr.file
+                    for aggr in self.site_info.attrs.fiat.aggregation
+                ]
+                attribute_names = [
+                    aggr.field_name for aggr in self.site_info.attrs.fiat.aggregation
+                ]
+                label_names = [
+                    f"Aggregation Label: {aggr.name}"
+                    for aggr in self.site_info.attrs.fiat.aggregation
+                ]
+
+                fa.apply_population_growth_new(
+                    population_growth=self.socio_economic_change.attrs.population_growth_new,
+                    ground_floor_height=self.socio_economic_change.attrs.new_development_elevation.value,
+                    elevation_type=self.socio_economic_change.attrs.new_development_elevation.type,
+                    area_path=area_path,
+                    ground_elevation=dem,
+                    aggregation_areas=aggregation_areas,
+                    attribute_names=attribute_names,
+                    label_names=label_names,
+                )
+
+            # Last apply population growth to existing objects
+            if self.socio_economic_change.attrs.population_growth_existing != 0:
+                fa.apply_population_growth_existing(
+                    population_growth=self.socio_economic_change.attrs.population_growth_existing,
                     ids=ids_existing,
                 )
-            else:
-                print("Impact measure type not recognized!")
 
-        # setup hazard for fiat
-        fa.set_hazard(self.hazard)
+            # Then apply Impact Strategy by iterating trough the impact measures
+            for measure in self.impact_strategy.measures:
+                if measure.attrs.type == "elevate_properties":
+                    fa.elevate_properties(
+                        elevate=measure,
+                        ids=ids_existing,
+                    )
+                elif measure.attrs.type == "buyout_properties":
+                    fa.buyout_properties(
+                        buyout=measure,
+                        ids=ids_existing,
+                    )
+                elif measure.attrs.type == "floodproof_properties":
+                    fa.floodproof_properties(
+                        floodproof=measure,
+                        ids=ids_existing,
+                    )
+                else:
+                    self.logger.warning(
+                        f"Impact measure type not recognized: {measure.attrs.type}"
+                    )
```

Comment:
Agreed, there will be when the fiat adapter is implemented according to the interface.

it will have (same for the sfincs adapter) a func `add_measure(self, measure: IMeasure)`
and the intelligence of which func to call will be in the fiat adapter